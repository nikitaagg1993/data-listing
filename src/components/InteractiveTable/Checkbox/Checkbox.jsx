/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import PropTypes from 'prop-types';

const Checkbox = ({
  checked,
  name,
  onClick,
}) => (
  <input
    type="checkbox"
    onClick={onClick}
    name={name}
    aria-label={name}
    checked={checked}
    onChange={() => null}
  />
);

Checkbox.propTypes = {
  name: PropTypes.string.isRequired,
  checked: PropTypes.bool,
  onClick: PropTypes.func,
};

Checkbox.defaultProps = {
  checked: false,
  onClick: null,
};

export default Checkbox;
