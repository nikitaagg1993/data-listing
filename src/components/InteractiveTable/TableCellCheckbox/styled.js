import styled from 'styled-components';
import { CellBase } from '../TableCell/styled';

const TableCellCheckboxStyle = styled(CellBase)`
  flex: 0 0 48px;
  justify-content: center;
  align-items: center;
  user-select: none;
  white-space: nowrap;
`;

export default TableCellCheckboxStyle;
