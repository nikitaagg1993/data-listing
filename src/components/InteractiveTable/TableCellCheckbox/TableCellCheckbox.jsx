import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import { useTableContext } from '../DataTableContext';
import Checkbox from '../Checkbox';
import TableCellCheckboxStyle from './styled';

const TableCellCheckbox = ({ name, row, selected }) => {
  const {
    dispatch,
    data,
    keyField,
  } = useTableContext();

  const handleOnRowSelected = useCallback(() => dispatch({
    type: 'SELECT_SINGLE_ROW',
    row,
    isSelected: selected,
    keyField,
    rowCount: data.length,
  }), [dispatch, row, selected, keyField, data.length]);

  return (
    <TableCellCheckboxStyle
      onClick={(e) => e.stopPropagation()}
      noPadding
    >
      <Checkbox
        name={name}
        checked={selected}
        aria-checked={selected}
        onClick={handleOnRowSelected}
      />
    </TableCellCheckboxStyle>
  );
};

TableCellCheckbox.propTypes = {
  name: PropTypes.string.isRequired,
  row: PropTypes.shape({}).isRequired,
  selected: PropTypes.bool.isRequired,
};

export default TableCellCheckbox;
