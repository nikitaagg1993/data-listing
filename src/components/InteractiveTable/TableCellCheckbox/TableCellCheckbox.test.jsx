import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import TableCellCheckbox from './TableCellCheckbox';
import Checkbox from '../Checkbox';

jest.mock('../DataTableContext', () => ({
  useTableContext: () => ({
    dispatch: jest.fn(),
    data: [],
  }),
}));

test('Table Cell Checkbox', () => {
  const row = { name: 'title' };

  const tree = renderer.create(<TableCellCheckbox
    name="Name"
    row={row}
    selected
  />);
  const testInstance = tree.root;

  expect(tree.toJSON()).toMatchSnapshot();
  expect(testInstance.findAllByType(Checkbox).length).toEqual(1);
  expect(testInstance.props).toMatchObject({ row: { name: 'title' } });
});
