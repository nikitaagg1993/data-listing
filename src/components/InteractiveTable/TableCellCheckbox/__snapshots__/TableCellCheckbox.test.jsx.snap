// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Cell Checkbox 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding: 0;
}

.c1 {
  -webkit-flex: 0 0 48px;
  -ms-flex: 0 0 48px;
  flex: 0 0 48px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
}

<div
  className="c0 c1"
  onClick={[Function]}
>
  <input
    aria-label="Name"
    checked={true}
    name="Name"
    onChange={[Function]}
    onClick={[Function]}
    type="checkbox"
  />
</div>
`;
